lib/ansible/plugins/loader.py

```diff
@@ -799,7 +801,8 @@ def get_with_context(self, name, *args, **kwargs):
 
         if path not in self._module_cache:
             self._module_cache[path] = self._load_module_source(name, path)
-            self._load_config_defs(name, self._module_cache[path], path)
+            self._load_config_defs(
+                name, self._module_cache[path], path, collection_name=plugin_load_context.plugin_resolved_collection)
```

Comment:
@bcoca including the collection name in `name` isn't trivial, since sometimes `name` is not a short name, but something else; for example I saw `ansible_collections.ansible.builtin.plugins.lookup.file` when running the following playbook:
```
- hosts: localhost
  tasks:
    - debug:
        msg: "{{ lookup('ansible.builtin.file', 'asdf') }}"
```
I guess that this shouldn't happen and should be fixed, but without that fixed I don't see how to combine name + collection name into one parameter without introducing a new syntax (that uses another separator).